components/feature/downloads/src/main/java/mozilla/components/feature/downloads/JobService.kt

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.downloads
+
+import android.app.Service
+import android.content.Intent
+import androidx.annotation.CallSuper
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+import java.util.concurrent.Executors
+
+/**
+ * Service that runs suspend functions in parallel.
+ * When all jobs are completed, the service is stopped automatically.
+ */
+abstract class JobService(
```

Comment:
nit: Let's try to find a different name for that. [JobService](https://developer.android.com/reference/android/app/job/JobService) is already used in the JobScheduler API and at first I thought you'd use it here. :)

Reply:
This looks pretty similar to [IntentService](https://developer.android.com/reference/android/app/IntentService)